### PR TITLE
Feature/cancel pending download

### PR DESCRIPTION
**Pull request by siepra**

### CODE REVIEW

File: packages/backend/src/storage/storage.ts [hunk 1]
```diff
@@ -506,29 +514,56 @@ export class Storage {
...
-      downloadProgress: downloadCompleted
-    }
+      // Canceled Download
+      this.io.updateDownloadProgress(statusCanceled)
 
-    // Downloaded file
-    this.io.updateDownloadProgress(statusCompleted)
+      // Clear cancellation signal
+      const index = this.downloadCancellations.indexOf(metadata.cid)
+      if (index > -1) {
```
Comment (Dinek007) [inline packages/backend/src/storage/storage.ts:535]: is it not better `index !== -1 ` ?

Comment (siepra) [reply packages/backend/src/storage/storage.ts:535]: `index > -1` is less tolerant also `index !== -1` is true for index equal undefined

Comment (Dinek007) [reply packages/backend/src/storage/storage.ts:535]: or replace 
`const index = this.downloadCancellations.indexOf(metadata.cid)
      if (index > -1) {`
with
`if(this.downloadCancellations.includes(metadata.cid)){`

File: packages/backend/src/storage/storage.ts [hunk 2]
```diff
@@ -506,29 +514,56 @@ export class Storage {
...
+        this.downloadCancellations.splice(index, 1)
```
Comment (Dinek007) [inline packages/backend/src/storage/storage.ts:536]: why splice form `index` to `1`?

File: packages/desktop/src/renderer/components/Channel/File/FileComponent/FileComponent.tsx [hunk 3]
```diff
@@ -212,11 +240,6 @@ export const FileComponent: React.FC<FileComponentProps & FileActionsProps> = ({
               color: theme.palette.colors.darkGray,
               icon: clockIconGray
             }}
-            // hover={{
```
Comment (Dinek007) [inline packages/desktop/src/renderer/components/Channel/File/FileComponent/FileComponent.tsx:215]: why we want to keep that code?

**Review by Dinek007 — APPROVED** [review]

File: packages/backend/src/storage/storage.ts (hunk 1 above)
Comment (siepra) [reply packages/backend/src/storage/storage.ts:535]: You need to know the index of an element in array to be able to remove it. So yeah, you can do it by using `includes` method but it'll only give you additional line of code.
Also it's always safe to assume that some value can be `undefined` or `null`, especially when it doesn't force you to change the underlaying logic

**Pull request merged**